lib/va_profile/models/email.rb

```diff
@@ -67,6 +67,17 @@ def self.build_from(body)
           va_profile_id: body['va_profile_id'] || body['vet360_id']
         )
       end
+
+      # Override the confirmation_date setter to correct it if it's after source_date.
+      # This prevents issues where client-provided dates may be ahead due to time differences.
+      # @param value [Time, String, nil] the confirmation date to set
+      # @return [Time, String, nil] the corrected confirmation date
+      def confirmation_date=(value)
+        @confirmation_date = value
+        return if @confirmation_date.blank? || source_date.blank?
+
+        @confirmation_date = source_date if @confirmation_date > source_date
```

Comment:
If the param can be a Time, String, or nil, is it possible that we end up comparing a Time to a string?

In other words, do we have a guarantee that `@confirmation_date` and `source_date` are the same value type at compare time? I'm unclear on whether the type conversions would happen before this setter method

Reply:
Good catch. 
> When we override the setters, we bypass the automatic type casting that Vets::Attributes provides
I tested this with some test code. So, I added explicit casting.